### PR TITLE
Fix deletion of CiliumNetworkPolicy

### PR DESCRIPTION
**Pull request by tgraf**
This PR addresses two issues:
 * The unique identification of CNP and NP was achieved with a combination of namespace and name of the policy. This is not unique enough across CNP and NP as both resource types can hold resources with identical names.
 * The deletion of CNP required the original rules to be embedded in order to extract the labels. This is not required and may have caused CNPs to not be properly deleted.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5556)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

EDIT: Weird CI failure that appears to be related to Jenkins itself:
```
Using context: Cilium-Ginkgo-Tests
an exception which occurred:
	in field com.cloudbees.groovy.cps.impl.BlockScopeEnv.locals
	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@2c3cd7aa
	in field com.cloudbees.groovy.cps.impl.CallEnv.caller
	in object com.cloudbees.groovy.cps.impl.FunctionCallEnv@19265812
	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
```

**Comment by tgraf** [comment]
test-me-please

Edited by Eloy, failed builds: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/6462/
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/6463/

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

EDIT: green build

File: pkg/k8s/apis/cilium.io/const.go [hunk 1]
```diff
@@ -18,9 +18,14 @@ const (
 	// PolicyLabelName is the name of the policy label which refers to the
 	// k8s policy name.
 	PolicyLabelName = "io.cilium.k8s.policy.name"
+
 	// PolicyLabelNamespace is the policy's namespace set in k8s.
 	PolicyLabelNamespace = "io.cilium.k8s.policy.namespace"
 
+	// PolicyLabelDerivedFrom is the resource type which was used to derie
```
Comment (ianvernon) [inline pkg/k8s/apis/cilium.io/const.go]: derive

File: daemon/k8s_watcher.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ import (
 	"github.com/cilium/cilium/pkg/ipcache"
```
Comment (ianvernon) [inline daemon/k8s_watcher.go:34]: Nit: typo in commit. "no rely" --> "not rely"

**Review by ianvernon — COMMENTED** [review]
Great find! I know this is WIP, but wanted to give it a look, and it looks like it is complete :)

**Comment by tgraf** [comment]
test-me-please

EDIT: green build

**Review by aanm — APPROVED** [review]

**Pull request merged**